### ISSUE TITLE
Kustomize bails out due to 'mapping key "enable-envoy-config" already defined' with gatewayAPI.enabled=true and ingressController.enabled=true

### ISSUE THREAD

**Issue by benpsnyder** [cilium/cilium#24226]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Currently I have to implement a workaround as I can'y have these settings in my `values.yaml` at the same time

```yaml
gatewayAPI:
  enabled: true
ingressController:
  enabled: true
```

![image](https://user-images.githubusercontent.com/707567/223480568-ff368777-8a26-45a4-87ce-62ff764c72b5.png)

### Cilium Version

1.13.0

### Kernel Version

Linux AZ-LXWEB03 5.15.0-1034-azure #41-Ubuntu SMP Fri Feb 10 19:59:45 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

### Kubernetes Version

1.26

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by sayboras** [comment]
it's fixed as part of https://github.com/cilium/cilium/pull/23866, and will be available in next patch release.

**Comment by sayboras** [comment]
As the fix is available, and back-ported to 1.13, closing this issue now. Feel free to let us know if there is anything else. Thanks.
